src/haddock/modules/topology/topocg/aa2cg.py

```diff
@@ -408,47 +431,51 @@ def determine_ss(structure, skipss, pdbf_path):
         pdbf_path:
 
     Returns:
+        structure:
 
     """
     # calculate SS
     for model in structure:
+        tmp_file_name = create_file_with_cryst(pdbf_path)
+
         if skipss:
             continue
         else:
             try:
-                dssp = DSSP(model, pdbf_path)
+                p = subprocess.Popen(["dssp", tmp_file_name, "--output-format", "dssp"], stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True)
+                dssp_raw, errors = p.communicate()
+                dssp_raw = dssp_raw.split('#')[1].split('\n')[1:-1]
             except:  # TODO: think about making this exception more specific
                 # no secondary structure detected for this model
                 log.warning('SS could not be assigned, assigning code 1 to all residues')
                 continue
+            finally:
+                if Path(tmp_file_name).exists(): Path(tmp_file_name).unlink()
+
+        dssp = {}
+
+        for line in dssp_raw:
+            if line[11:12] in dssp.keys():
+                dssp[line[11:12]].append(line[16:17])
+            else:
+                dssp[line[11:12]] = [line[16:17]]
```

Comment:
you can remove this whole if/else block if you use the [`setdefault`](https://docs.python.org/3.12/library/stdtypes.html#dict.setdefault) method, and by doing `line[11:12]` what you are actually doing is `line[11]`, so you can also cast it directly to variables:

```python
for line in dssp_raw:
    var_a, var_b = line[11], line[16]
    dssp.setdefault(var_a, []).append(var_b)
```